agent/api/container.go

```diff
@@ -216,6 +221,45 @@ func (c *Container) GetKnownExitCode() *int {
 	return c.knownExitCode
 }
 
+// SetRegistryAuthCredentials will set the credentials for pulling image from ecr
+func (c *Container) SetRegistryAuthCredentials(credential *credentials.IAMRoleCredentials) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.RegistryAuthentication == nil || c.RegistryAuthentication.ECRAuthData == nil {
+		return fmt.Errorf("Container: set ecr pull credentials failed, no registry auth data for container")
+	}
+
+	if !c.RegistryAuthentication.ECRAuthData.UseExecutionRole {
+		return fmt.Errorf("Container: set ecr pull credentials failed, using task level credentials is disabled")
+	}
+
+	c.RegistryAuthentication.ECRAuthData.PullCredentials = *credential
+	return nil
+}
+
+// SetupExecutionRoleFlag will check if the container is configured to use
+// task execution role for pulling
+func (c *Container) SetupExecutionRoleFlag() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.RegistryAuthentication != nil &&
+		c.RegistryAuthentication.Type == "ecr" &&
+		c.RegistryAuthentication.ECRAuthData != nil &&
+		c.RegistryAuthentication.ECRAuthData.UseExecutionRole {
+		c.PullWithExecutionRole = true
```

Comment:
Do you really need the `PullWithExecutionRole` flag? Can't you always compute it when you need it?